xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -440,3 +419,318 @@ bool CEnumeratorHD::IsSDRSupported()
+  result.reserve(formats.m_output.size());
+  for (const auto& format : formats.m_output)
+  {
+    const std::string name = DX::DXGIFormatToString(format);
+
+    if (name.find('R') != std::string::npos && name.find('G') != std::string::npos &&
+        name.find('B') != std::string::npos)
+    {
+      result.emplace_back(format);
+    }
+  }
+
+  return result;
+}
+
+bool CEnumeratorHD::CheckConversion(DXGI_FORMAT inputFormat,
+                                    DXGI_COLOR_SPACE_TYPE inputCS,
+                                    DXGI_FORMAT outputFormat,
+                                    DXGI_COLOR_SPACE_TYPE outputCS)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+  return CheckConversionInternal(inputFormat, inputCS, outputFormat, outputCS);
+}
+
+bool CEnumeratorHD::CheckConversionInternal(DXGI_FORMAT inputFormat,
+                                            DXGI_COLOR_SPACE_TYPE inputCS,
+                                            DXGI_FORMAT outputFormat,
+                                            DXGI_COLOR_SPACE_TYPE outputCS) const
+{
+  // Not initialized yet or OS < Windows 10
+  if (!m_pEnumerator || !m_pEnumerator1)
+    return false;
+
+  HRESULT hr;
+  BOOL supported;
+
+  if (SUCCEEDED(hr = m_pEnumerator1->CheckVideoProcessorFormatConversion(
+                    inputFormat, inputCS, outputFormat, outputCS, &supported)))
+  {
+    return (supported == TRUE);
+  }
+  else
+  {
+    CLog::LogF(LOGERROR, "unable to validate the format conversion, error {}",
+               DX::GetErrorDescription(hr));
+    return false;
+  }
+}
+
+ProcessorConversions CEnumeratorHD::ListConversions(
+    DXGI_FORMAT inputFormat,
+    const std::vector<DXGI_COLOR_SPACE_TYPE>& inputColorSpaces,
+    const std::vector<DXGI_FORMAT>& outputFormats,
+    const std::vector<DXGI_COLOR_SPACE_TYPE>& outputColorSpaces) const
+{
+  // Not initialized yet or OS < Windows 10
+  if (!m_pEnumerator || !m_pEnumerator1)
+    return {};
+
+  ProcessorConversions result;
+
+  for (const DXGI_COLOR_SPACE_TYPE& inputCS : inputColorSpaces)
+  {
+    for (const DXGI_FORMAT& outputFormat : outputFormats)
+    {
+      for (const DXGI_COLOR_SPACE_TYPE& outputCS : outputColorSpaces)
+      {
+        if (CheckConversionInternal(inputFormat, inputCS, outputFormat, outputCS))
+        {
+          result.push_back(
+              ProcessorConversion{m_input_dxgi_format, inputCS, outputFormat, outputCS});
+        }
+      }
+    }
+  }
+  return result;
+}
+
+void CEnumeratorHD::LogSupportedConversions(const DXGI_FORMAT& inputFormat,
+                                            const DXGI_COLOR_SPACE_TYPE heuristicsInputCS,
+                                            const DXGI_COLOR_SPACE_TYPE inputNativeCS,
+                                            const DXGI_FORMAT& outputFormat,
+                                            const DXGI_COLOR_SPACE_TYPE heuristicsOutputCS,
+                                            const VideoPicture& picture)
```

Comment:
Unused parameter  `const VideoPicture& picture`

Reply:
yes, leftover of intermediate version